### PR TITLE
Make CI testing green again 

### PR DESCRIPTION
**Pull request by ylobankov**
This patch contains a series of fixes to make CI testing green again:

1. [test-run: bump to new version](https://github.com/tarantool/tarantool/pull/7090/commits/6b00637bf766a54bb7edaed190c84fc1a6e8c785)

2. [ci: change value for VARDIR env variable to /tmp/t](https://github.com/tarantool/tarantool/pull/7090/commits/e5eaca167add7f92ac8ab23539ac5ad6cda19aa1)

3. [ci: add VARDIR to PRESERVE_ENVVARS env variable](https://github.com/tarantool/tarantool/pull/7090/commits/c04deb45a3260a6da19a5ce21b3e50e53a9c0a3b)

4. [luatest: use /tmp/t as default socket dir](https://github.com/tarantool/tarantool/pull/7090/commits/337c037627baeede1c62db29bc57b211eda18dcf)

5. [cmake: drop --vardir option when running tests](https://github.com/tarantool/tarantool/pull/7090/commits/cbe7968edf0abb2472266d1203f6a0e4dfe7ada7)

### CODE REVIEW

**Review by Totktonada — APPROVED** [review]
No objections from me.

Please, don't forget to cherry-pick changes to 1.10, where applicable, to keep the testing infrastructure more or less same.

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -117,11 +117,12 @@ jobs:
...
-    needs: tarantool
-    uses: tarantool-php/client/.github/workflows/reusable_qa.yml@master
-    with:
-      artifact_name: tarantool-ubuntu-focal-${{ github.sha }}
+# TODO: Enable testing when tarantool-php/client/issues/79 is resolved.
+#  php-client:
+#    needs: tarantool
+#    uses: tarantool-php/client/.github/workflows/reusable_qa.yml@master
+#    with:
+#      artifact_name: tarantool-ubuntu-focal-${{ github.sha }}
```
Comment (Totktonada) [inline .github/workflows/integration.yml:125]: https://github.com/tarantool-php/client/issues/79 is fixed.

**Review by locker — APPROVED** [review]

File: .github/workflows/integration.yml (hunk 1 above)
Comment (ylobankov) [reply .github/workflows/integration.yml]: Fixed.

**Comment by ylobankov** [comment]
Commit with disabling tarantool-php/client tests was removed as https://github.com/tarantool-php/client/issues/79 is resolved.

**Comment by locker** [comment]
Cherry-picked to 2.10.

**Pull request merged**